pkg/agent/secondarynetwork/podwatch/controller_test.go

```diff
@@ -0,0 +1,399 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podwatch
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/golang/mock/gomock"
+	netdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	netdefclientfake "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/fake"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/util/workqueue"
+
+	"antrea.io/antrea/pkg/agent/secondarynetwork/cnipodcache"
+	ipamtesting "antrea.io/antrea/pkg/agent/secondarynetwork/ipam/testing"
+	podwatchtesting "antrea.io/antrea/pkg/agent/secondarynetwork/podwatch/testing"
+)
+
+const (
+	testNamespace = "nsA"
+	testNode      = "test-node"
+
+	netAttachConfig = `{
+    "cniVersion": "0.3.0",
+    "type": "antrea",
+    "mode": "sriov",
+    "ipam": {
+        "type": "whereabouts",
+        "datastore": "kubernetes",
```

Comment:
@antoninbas its required to provide the kubeconfig path at the network CRD ipam for whereabouts to talk to the cluster. please update the controller_test.go 
        "kubernetes": {
          "kubeconfig": "/host/etc/cni/net.d/whereabouts.d/whereabouts.kubeconfig"
        },

Reply:
I'm adding it, but note that this `ipam` information is not actually used by the unit tests, since we use a mock IPAM object, and we don't actually invoke the whereabouts plugin